packages/wallet/src/test-utils/launchNode.ts

```diff
@@ -0,0 +1,177 @@
+import { BaseAssetId } from '@fuel-ts/address/configs';
+import { Provider } from '@fuel-ts/providers';
+import { spawn } from 'child_process';
+import fsSync from 'fs';
+import fs from 'fs/promises';
+import os from 'os';
+import path from 'path';
+import { getPortPromise } from 'portfinder';
+import kill from 'tree-kill';
+
+import type { WalletUnlocked } from '../wallets';
+
+import { defaultChainConfig } from './defaultChainConfig';
+import { generateTestWallet } from './generateTestWallet';
+
+const defaultFuelCoreArgs = ['--vm-backtrace', '--utxo-validation', '--manual_blocks_enabled'];
+
+type LaunchNodeOptions = {
+  chainConfigPath: string;
+  consensusKey: string;
+  ip?: string;
+  port?: string;
+  args?: string[];
+  useSystemFuelCore?: boolean;
+};
+
+/**
+ * Launches a fuel-core node.
+ * @param chainConfigPath - path to the chain configuration file.
+ * @param consensusKey - the consensus key to use.
+ * @param ip - the ip to bind to. (optional, defaults to 0.0.0.0)
+ * @param port - the port to bind to. (optional, defaults to 4000 or the next available port)
+ * @param args - additional arguments to pass to fuel-core
+ * @param useSystemFuelCore - whether to use the system fuel-core binary or the one provided by the \@fuel-ts/fuel-core package.
+ * */
+export const launchNode = async ({
+  chainConfigPath,
+  consensusKey,
+  ip,
+  port,
+  args = defaultFuelCoreArgs,
+  useSystemFuelCore = false,
+}: LaunchNodeOptions): Promise<{
+  cleanup: () => void;
+  ip: string;
+  port: string;
+}> =>
+  // eslint-disable-next-line no-async-promise-executor
+  new Promise(async (resolve) => {
+    // This string is logged by the client when the node has successfully started. We use it to know when to resolve.
+    const graphQLStartSubstring = 'Binding GraphQL provider to';
+
+    const command = useSystemFuelCore ? 'fuel-core' : './node_modules/.bin/fuels-core';
+
+    const ipToUse = ip || '0.0.0.0';
+
+    const portToUse =
+      port ||
+      (
+        await getPortPromise({
+          port: 4000, // tries 4000 first, then 4001, then 4002, etc.
+        })
+      ).toString();
+
+    const child = spawn(command, [
+      'run',
+      '--ip',
+      ipToUse,
+      '--port',
+      portToUse,
+      '--db-type',
+      'in-memory',
+      '--consensus-key',
+      consensusKey,
+      '--chain',
+      chainConfigPath,
+      ...args,
+    ]);
+
+    // Cleanup function where fuel-core is stopped.
+    const cleanup = () => {
+      kill(Number(child.pid));
```

Comment:
We could only if we wanted to re-implement the process tree traversal by hand.

Reply:
I see your point @camsjams but I chose to use this package instead of vanilla Node.js because this package goes also kills any child process spawned by the process we are trying to kill, like @arboleya mentioned. I believe Anderson is also using the same package in the contracts tooling PR

Reply:
FWIW, the source and tests look concise, and the package is massively [used](https://www.npmjs.com/package/tree-kill).

Considering it is still in vogue, the fact it is this old might tell me it is really good.

The downloads/issues ratio also transmits the feeling of an outstanding solid package.

Reply:
Not in the tooling PR, because there it's not aimed at tests.

Reply:
Can we make it dev dependency in this PR too? Won't we need this logic at run-time?